src/message_interface/response_type.rs

```diff
@@ -0,0 +1,130 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::HashSet;
+
+use crate::{
+    api::PreparedTransactionData, builder::NetworkInfo, node_api::high_level::AddressBalance, node_manager::node::Node,
+    Error, NodeInfoWrapper,
+};
+
+use bee_message::{address::Address, input::UtxoInput, output::OutputId, Message, MessageId};
+use bee_rest_api::types::{
+    dtos::{PeerDto, ReceiptDto},
+    responses::{
+        InfoResponse as NodeInfo, MessageMetadataResponse, MilestoneResponse, OutputResponse, RentStructureResponse,
+        TreasuryResponse, UtxoChangesResponse as MilestoneUTXOChanges,
+    },
+};
+
+use serde::Serialize;
+
+/// The response message.
+#[derive(Serialize, Debug)]
+#[serde(tag = "type", content = "payload")]
+pub enum ResponseType<'a> {
+    /// GenerateAddress response.
+    GeneratedAddresses(Vec<String>),
+    /// Generated message
+    GeneratedMessage(PreparedTransactionData),
+    /// Node
+    Node(Node),
+    // /// Proof of work provider
+    // PoWProvider(ClientMiner),
```

Comment:
The client miner is only required internally so we can set the parameters from the PoW, but for bindings it's useless
```suggestion
```